### PR TITLE
NMS-12505: Provide ID and IpInterface in JSON results

### PR DESCRIPTION
**Pull request by christianpape**
* JIRA: https://issues.opennms.org/browse/NMS-12505

### CODE REVIEW

File: opennms-model/src/main/java/org/opennms/netmgt/model/OnmsMonitoredService.java [hunk 1]
```diff
@@ -416,14 +421,20 @@ public void removeMetaData(final String context) {
...
     @XmlElement(name="ipInterfaceId")
     @ManyToOne(optional=false, fetch=FetchType.LAZY)
     @JoinColumn(name="ipInterfaceId")
     public OnmsIpInterface getIpInterface() {
         return m_ipInterface;
     }
 
+    @JsonProperty("ipInterfaceId")
+    @Transient
+    public Integer getFoo() {
```
Comment (mvrueden) [inline opennms-model/src/main/java/org/opennms/netmgt/model/OnmsMonitoredService.java]: Fu und Fara oder was?

Should probably be named differently :D

File: opennms-model/src/main/java/org/opennms/netmgt/model/OnmsMonitoredService.java [hunk 2]
```diff
@@ -416,14 +421,20 @@ public void removeMetaData(final String context) {
...
+        return m_ipInterface.getId();
```
Comment (mvrueden) [inline opennms-model/src/main/java/org/opennms/netmgt/model/OnmsMonitoredService.java]: Can `m_ipInterface` ever be null?

File: opennms-model/src/main/java/org/opennms/netmgt/model/OnmsMonitoredServiceDetail.java [hunk 3]
```diff
@@ -73,6 +75,7 @@ public OnmsMonitoredServiceDetail(OnmsMonitoredService service) {
         m_statusCode = service.getStatus();
         m_status = service.getStatusLong();
         m_id = service.getXmlId();
+        m_ipInterfaceId = service.getIpInterface().getId();
```
Comment (mvrueden) [inline opennms-model/src/main/java/org/opennms/netmgt/model/OnmsMonitoredServiceDetail.java]: Do we need to store this as a field, can't we just return that in the `getIpinterfaceId()` method?

File: opennms-model/src/main/java/org/opennms/netmgt/model/OnmsMonitoredServiceDetail.java [hunk 4]
```diff
@@ -140,6 +143,11 @@ public void setId(String id) {
         m_id = id;
     }
 
+    @XmlAttribute(name="ipInterfaceId")
+    public Integer getIpinterfaceId() {
```
Comment (mvrueden) [inline opennms-model/src/main/java/org/opennms/netmgt/model/OnmsMonitoredServiceDetail.java]: Should be `getIpInterfaceId()` instead.

File: opennms-model/src/main/java/org/opennms/netmgt/model/OnmsMonitoredService.java (hunk 1 above)
Comment (christianpape) [reply opennms-model/src/main/java/org/opennms/netmgt/model/OnmsMonitoredService.java]: Yes, of course.

File: opennms-model/src/main/java/org/opennms/netmgt/model/OnmsMonitoredService.java (hunk 2 above)
Comment (christianpape) [reply opennms-model/src/main/java/org/opennms/netmgt/model/OnmsMonitoredService.java]: I don't think so.

File: opennms-model/src/main/java/org/opennms/netmgt/model/OnmsMonitoredServiceDetail.java (hunk 3 above)
Comment (christianpape) [reply opennms-model/src/main/java/org/opennms/netmgt/model/OnmsMonitoredServiceDetail.java]: done

File: opennms-model/src/main/java/org/opennms/netmgt/model/OnmsMonitoredServiceDetail.java (hunk 4 above)
Comment (christianpape) [reply opennms-model/src/main/java/org/opennms/netmgt/model/OnmsMonitoredServiceDetail.java]: done

**Pull request merged**